cli/CommandLineParser.php

```diff
@@ -0,0 +1,277 @@
+<?php
+declare(strict_types=1);
+
+class CommandLineParser {
+	/** @var array<string,Option> */
+	private array $options;
+	/** @var array<string,array{defaultInput:?string[],required:?bool,aliasUsed:?string,values:?string[]}> */
+	private array $inputs;
+
+	/** Adds an option that produces an error message if not set. */
+	public function addRequiredOption(string $name, Option $option): static {
+		$this->inputs[$name] = [
+			'defaultInput' => null,
+			'required' => true,
+			'aliasUsed' => null,
+			'values' => null,
+		];
+		$this->options[$name] = $option;
+
+		return $this;
+	}
+
+	/**
+	 * Adds an optional option.
+	 * @param string $defaultInput If not null this value is received as input in all cases where no
+	 *  user input is present. e.g. set this if you want an option to always return a value.
+	 */
+	public function addOption(string $name, Option $option, string $defaultInput = null) :static {
+		$this->inputs[$name] = [
+			'defaultInput' => is_string($defaultInput) ? [$defaultInput] : $defaultInput,
+			'required' => null,
+			'aliasUsed' => null,
+			'values' => null,
+		];
+		$this->options[$name] = $option;
+
+		return $this;
+	}
+
+	/**
+	 * @template T
+	 * @param class-string<T> $target
+	 * @return T
+	 */
+	public function parse(string $target) {
```

Comment:
This function seems to be always called with `stdClass::class`, isn't it?
The parameter could probably be removed then, to make the calls simpler (`$parser->parse(stdClass::class)` is not very obvious)

Reply:
See https://github.com/FreshRSS/FreshRSS/pull/6099#discussion_r1494907030

Reply:
Simplified in https://github.com/KasimirCash/FreshRSS/pull/1